### ISSUE TITLE
builds require Internet for tools/java and tools/jmx

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylla#6894]
Since 5d99d667ec7de8cb2d0bb514b290fa3f4a0c892e, builds now contact the maven repositories to fetch Java artifacts. This is slow and also makes builds less reliable.

**Comment by avikivity** [comment]
We might prime the cache by downloading all the dependencies in ./install-depedencies.sh. We'll need to stop mounting ~/.m2 in dbuild, since maven supports only one cache.

**Comment by slivne** [comment]
@avikivity is this a regression ? I think scylla-tools-java was always built this way

**Comment by avikivity** [comment]
It's not a regression (it is for scylla.git since we started building the submodules, but not overall).

**Comment by penberg** [comment]
Good idea, but it's not clear to me how we should implement this. With Maven, we could look into using `mvn dependency:go-offline` in `install-dependencies.sh` maybe? There's probably something similar for the `ant` build too.

**Comment by avikivity** [comment]
Yes, that's what I thought to do. I didn't figure out the exact sequence.

**Comment by avikivity** [comment]
Fixed by 5e1cf90a51b1c3df146f67d141bee62ba2a65f62 (but Rust builds require the Internet instead)

**Comment by mykaul** [comment]
Rust should be conditionally build. It's only for UDF, no?
I also do believe sccache (but not ccache) can cache also Rust compilation.

**Comment by avikivity** [comment]
> Rust should be conditionally build. 

Why? Is the build and test system not complicated enough?

> It's only for UDF, no? 

Yes.

> I also do believe sccache (but not ccache) can cache also Rust compilation.

It's not a major component in build time.

**Comment by avikivity** [comment]
Closing as won't fix, this is a hopeless cause in this day and age.

While we no longer have tools/java, python (tools/cqlsh) and rust (wasm and vector-store (?!)) are hopelessly addicted to the internet.
